shared/health/src/main/java/io/pravega/shared/health/InlineHealthIndicator.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Dell Inc., or its subsidiaries. All Rights Reserved.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.shared.health;
+
+import lombok.NonNull;
+
+import java.util.function.BiConsumer;
+
+/**
+ * {@link InlineHealthIndicator} provides a syntactic alternative to the {@link HealthIndicator}, but is ultimately functionally
+ * equivalent. Instead of creating a {@link HealthIndicator} by subclassing it and implementing {@link HealthIndicator#doHealthCheck(Health.HealthBuilder)},
+ * this simply provides the option to supply that abstract method as a lambda expression.
+ */
+public class InlineHealthIndicator extends HealthIndicator {
+
+    private final BiConsumer<Health.HealthBuilder, DetailsProvider> doHealthCheck;
+
+    @NonNull
+    InlineHealthIndicator(String name, BiConsumer<Health.HealthBuilder, DetailsProvider> doHealthCheck) {
```

Comment:
Is this meant to be `public`? Otherwise nobody from outside your package can create this class.

Reply:
Correct.